<a href="https://colab.research.google.com/github/saerarawas/AAI_634O_A11_202520/blob/main/week1/Sales_dashboard.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [43]:
import pandas as pd
import numpy as np

# Define the number of records
num_records = 100

# Generate random dates within a specific range
dates = pd.date_range(start='2023-01-01', end='2023-12-31', periods=num_records)

# Generate product IDs (e.g., P001, P002, ..., P100)
product_ids = [f'P{str(i).zfill(3)}' for i in range(1, num_records + 1)]

# Generate random sales amounts
sales_amounts = np.random.randint(100, 1000, size=num_records)

# Define a list of store locations
store_locations = ['New York', 'Los Angeles', 'Chicago', 'Houston', 'Phoenix']

# Randomly assign store locations to the records
locations = np.random.choice(store_locations, size=num_records)

# Create a DataFrame with the generated data
data = {
    'date': dates,
    'product_id': product_ids,
    'sales_amount': sales_amounts,
    'store_location': locations
}

df = pd.DataFrame(data)

# Save the DataFrame to a CSV file
df.to_csv('sales_data.csv', index=False)

# Display the first few rows of the DataFrame
print(df.head())

                           date product_id  sales_amount store_location
0 2023-01-01 00:00:00.000000000       P001           346        Phoenix
1 2023-01-04 16:14:32.727272727       P002           997        Houston
2 2023-01-08 08:29:05.454545454       P003           815        Houston
3 2023-01-12 00:43:38.181818181       P004           931        Phoenix
4 2023-01-15 16:58:10.909090909       P005           923        Chicago


In [44]:
import pandas as pd

# Load sales data from CSV
sales_data = pd.read_csv('sales_data.csv')

# Display the first few rows of the dataset
print(sales_data.head())

# Calculate total sales
total_sales = sales_data['sales_amount'].sum()
print(f'Total Sales: {total_sales}')

# Calculate sales by location
sales_by_location = sales_data.groupby('store_location')['sales_amount'].sum().reset_index()
print('Sales by Location:')
print(sales_by_location)

# Calculate sales by product
sales_by_product = sales_data.groupby('product_id')['sales_amount'].sum().reset_index()
print('Sales by Product:')
print(sales_by_product)

# Basic statistics
statistics = sales_data.describe()
print('Basic Statistics:')
print(statistics)

# Identify the store with the highest sales
highest_sales_store = sales_by_location.loc[sales_by_location['sales_amount'].idxmax()]
print(f'Highest Sales Store: {highest_sales_store}')

# Identify the product with the highest sales
highest_sales_product = sales_by_product.loc[sales_by_product['sales_amount'].idxmax()]
print(f'Highest Sales Product: {highest_sales_product}')



                            date product_id  sales_amount store_location
0  2023-01-01 00:00:00.000000000       P001           346        Phoenix
1  2023-01-04 16:14:32.727272727       P002           997        Houston
2  2023-01-08 08:29:05.454545454       P003           815        Houston
3  2023-01-12 00:43:38.181818181       P004           931        Phoenix
4  2023-01-15 16:58:10.909090909       P005           923        Chicago
Total Sales: 54967
Sales by Location:
  store_location  sales_amount
0        Chicago         11249
1        Houston         12138
2    Los Angeles         12299
3       New York          5882
4        Phoenix         13399
Sales by Product:
   product_id  sales_amount
0        P001           346
1        P002           997
2        P003           815
3        P004           931
4        P005           923
..        ...           ...
95       P096           866
96       P097           741
97       P098           109
98       P099           717
99       P10

In [45]:
!pip install dash
import dash
from dash import dcc, html
from dash.dependencies import Input, Output
import plotly.express as px
import pandas as pd

# Load sales data from CSV
sales_data = pd.read_csv('sales_data.csv')

# Aggregate sales data by store location
sales_by_location = sales_data.groupby('store_location')['sales_amount'].sum().reset_index()

# Initialize the Dash app
app = dash.Dash(__name__)

# Create a bar chart
fig = px.bar(sales_by_location, x='store_location', y='sales_amount', title='Sales by Store Location')

# Set up the layout of the app
app.layout = html.Div(children=[
    html.H1(children='Sales Dashboard'),
    dcc.Graph(figure=fig)
])

if __name__ == '__main__':
    app.run_server(debug=True)


<IPython.core.display.Javascript object>

In [46]:
import dash
from dash import dcc, html
from dash.dependencies import Input, Output
import plotly.express as px
import pandas as pd

# Load sales data from CSV
sales_data = pd.read_csv('sales_data.csv')

# Initialize the Dash app
app = dash.Dash(__name__)

# Function to aggregate sales data by store location
def aggregate_sales_by_location():
    return sales_data.groupby('store_location')['sales_amount'].sum().reset_index()

# Function to aggregate sales data by product
def aggregate_sales_by_product():
    return sales_data.groupby('product_id')['sales_amount'].sum().reset_index()

# Layout of the app
app.layout = html.Div([
    html.H1('Sales Dashboard'),

    # Dropdown for selecting the type of aggregation
    dcc.Dropdown(
        id='select-type',
        options=[
            {'label': 'Store Location', 'value': 'location'},
            {'label': 'Product', 'value': 'product'}
        ],
        value='location',  # Default value
        style={'width': '50%'}
    ),

    # Graph for visualizing total sales
    dcc.Graph(id='bar-chart')
])

# Callback to update the bar chart based on the selected type
@app.callback(
    Output('bar-chart', 'figure'),
    [Input('select-type', 'value')]
)
def update_bar_chart(selected_type):
    if selected_type == 'location':
        sales_by_location = aggregate_sales_by_location()
        fig = px.bar(sales_by_location, x='store_location', y='sales_amount', title='Sales by Store Location')
    else:
        sales_by_product = aggregate_sales_by_product()
        fig = px.bar(sales_by_product, x='product_id', y='sales_amount', title='Sales by Product')
    return fig

if __name__ == '__main__':
    app.run_server(debug=True)



<IPython.core.display.Javascript object>

Highest Sales Store:
store_location  :  Phoenix;

sales_amount  =        13399

Highest Sales Product:
product_id    :  P002;

sales_amount   = 997